caffe2/CMakeLists.txt

```diff
@@ -63,6 +63,7 @@ if(INTERN_BUILD_ATEN_OPS)
   set(CMAKE_POSITION_INDEPENDENT_CODE ${__caffe2_CMAKE_POSITION_INDEPENDENT_CODE})
 
   # Generate the headers wrapped by our operator
+  file(GLOB_RECURSE all_python "${PROJECT_SOURCE_DIR}/tools/codegen/*.py")
```

Comment:
That doesn't contain the autograd codegen? Is that on pupose?
Also this folder is nested, should this be `/**/*.py`?

Reply:
>That doesn't contain the autograd codegen? Is that on pupose?

Yes, the generator only actually uses `CodeTemplate` which is in `tools/codegen/`.  
https://github.com/pytorch/pytorch/blob/master/caffe2/contrib/aten/gen_op.py

>Also this folder is nested, should this be /**/*.py?

`ninja` confirms this includes recursive directories.
```
$ ninja -C build -t query /home/peter/git/pytorch/build/caffe2/contrib/aten/aten_op.h
/home/peter/git/pytorch/build/caffe2/contrib/aten/aten_op.h:
  input: CUSTOM_COMMAND
    ../tools/codegen/__init__.py
    ../tools/codegen/api/__init__.py
    ../tools/codegen/api/autograd.py
    ../tools/codegen/api/cpp.py
    ../tools/codegen/api/dispatcher.py
    ../tools/codegen/api/functionalization.py
    ../tools/codegen/api/meta.py
    ../tools/codegen/api/native.py
    ../tools/codegen/api/python.py
    ../tools/codegen/api/structured.py
    ../tools/codegen/api/translate.py
    ../tools/codegen/api/types.py
    ../tools/codegen/code_template.py
    ../tools/codegen/context.py
    ../tools/codegen/dest/__init__.py
    ../tools/codegen/dest/native_functions.py
    ../tools/codegen/dest/register_dispatch_key.py
    ../tools/codegen/gen.py
    ../tools/codegen/gen_backend_stubs.py
    ../tools/codegen/gen_functionalization_type.py
    ../tools/codegen/local.py
    ../tools/codegen/model.py
    ../tools/codegen/selective_build/__init__.py
    ../tools/codegen/selective_build/operator.py
    ../tools/codegen/selective_build/selector.py
    ../tools/codegen/utils.py
    aten/src/ATen/Declarations.yaml
    ../caffe2/contrib/aten/gen_op.py
    ../caffe2/contrib/aten/aten_op_template.h
  outputs:
```